File: src/js/utils/Node.js
class LinkedNode {
    constructor(val, next) {
        this.val = val
        this.next = (next === undefined ? null : next)
    }
}

class DoubleLinkedNode {
    constructor(val, prev, next) {
        this.val = val
        this.prev = (prev === undefined ? null : prev)
        this.next = (next === undefined ? null : next)
    }
}

File: src/py/utils/Node.py
class LinkedNode:
    def __init__(self, val):
        self.val = val
        self.next: LinkedNode | None = None

class DoubleLinkedNode:
    def __init__(self, val):
        self.val = val
        self.prev: DoubleLinkedNode | None = None
        self.next: DoubleLinkedNode | None = None


class TreeNode:
    def __init(self, val):
        self.val = val
        self.left: TreeNode | None = None
        self.right: TreeNode | None = None
